File: client/src/lib/api/banks.ts
import { apiClient } from "./client";
import { API_ENDPOINTS } from "./config";

export interface Bank {
  name: string;
  code: string;
  bin: string;
  shortName: string;
  logo: string;
  transferSupported: number;
  lookupSupported: number;
  support: number;
  isTransfer: number;
  swiftCode: string;
  countryCode: string;
  countryName: string;
}

export interface GetBanksResponse {
  status: string;
  data: Bank[];
}

export const getBanks = async (countryCode?: string): Promise<GetBanksResponse> => {
  try {
    const response = await apiClient.get<GetBanksResponse>(API_ENDPOINTS.banks, {
      params: {
        country_code: countryCode,
      },
    });
    return response.data;
  } catch (error) {
    throw error;
  }
};

File: spec/factories/banks.rb
FactoryBot.define do
  factory :bank do
    association :country
    sequence(:name) { |n| "Bank #{n}" }
    sequence(:code) { |n| "B#{n.to_s.rjust(2, '0')}" }
    sequence(:bin) { |n| "BIN#{n.to_s.rjust(3, '0')}" }
    short_name { "ShortBank" }
    logo { "https://example.com/logo.png" }
    transfer_supported { true }
    lookup_supported { true }
    support { 3 }
    is_transfer { true }
    swift_code { "SWIFT123" }
  end
end
